server_modules/scores.js

```diff
@@ -15,6 +16,29 @@ function reduceToMap(key) {
     }
 }
 
+function changeScores(callback) {
+    var path = fileSystem.getDataFilePath('scores.json');
+    return fileSystem.readJsonFile(path)
+    .then(function(data) {
+        return data;
+    }, function() {
+        return { version:2, scores: [] };
+    })
+    .then(callback)
+    .then(function(scores) {
+        lockfile.lock('scores.json.lock', { retries: 5, retryWait: 100 }, function (err) {
```

Comment:
There is a race-condition here! You should obtain the lock before you read and modify it.